### ISSUE TITLE
No object detected in find_object_3d.launch with Kinect 1

### ISSUE THREAD

**Issue by Severn-Vergil** [introlab/find-object#25]
Hi,
I use Kinect 1 and try to detect my stool in 1m, and it can not detected even after I move it in 0.5m. I find that the frequency of camera in find_object gui is 0Hz. I didn't modify the find_object_3d launch file. I've tried your robot-mapping with find-object example, with the help of rosbag play demo_find_object.bag. In the example object_8 is detected. Could you give me some advice about how to run the example in the real environment successfully?
Thank you in advance!
![Uploading Screenshot from 2016-09-11 20:24:02.png…]()

**Comment by matlabbe** [comment]
Hi,

The stool can be quite difficult to detect with visual features, unless the camera position is the same than when the object's image was taken. In the [robot mapping with Find-Object](https://www.youtube.com/watch?v=o1GSQanY-Do) example, the objects are [book covers](https://github.com/introlab/rtabmap_ros/tree/master/launch/data/books), which have a lot of distinctive visual features. The choice of features used may also influence at which scale/rotation/point of view an object can be detected. 

Note that with the stool, it may also be difficult to compute the depth of it (is it in metal? depth on such objects may be null) as it is not plain (holes). Find-Object 3D uses a [very simple approach](https://github.com/introlab/find-object/blob/48e6ea0147da79b0bdc59886c1c9ffe89998dc9c/src/ros/FindObjectROS.cpp#L69) to determine the depth of the object, using the center of it. If the center is empty, object could be detected but 3D pose not.

cheers

**Comment by Severn-Vergil** [comment]
Thank you for your useful advice! Is it means that in find_object_3d example, I have to provide an example image of the object for the launch file? I'm trying to find out the object detection part of the find_object source code.

**Comment by matlabbe** [comment]
Yes, you must set the [objects_path](https://github.com/introlab/rtabmap_ros/blob/238e78bae0844ef5f20f066ed8b12ae87ad513b6/launch/demo/demo_find_object.launch#L95) with objects that you want to detect.
